site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,177 @@
+
+At this point, we have created the `ApplicationContext` to manage state, an `Injector` to inject state into the application's widgets, and `Containers` to manage how properties and children from the injector and parent widgets are combined. In the next section, we will integrate these components into our application.
+
+{% section %}
+
+## Using state containers
+
+{% task 'Integrate containers into an application.' %}
+
+As mentioned in the previous section `Containers` are specialized form of widget since they descend from `WidgetBase`. Given that, they can be used just like any other widget. In our demo application, we can take advantage of that fact by simply replacing the `WorkerForm` and `WorkerContainer` with their container equivalents.
+
+{% instruction 'Replace the imports in the `App` module with the following.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:1-6 %}
+
+There are two major changes to the `App` module's imports. First, the widgets (`WorkerForm` and `WorkerContainer`) have been replaced by their container equivalents (`WorkerFormContainer` and `WorkerContainerContainer`). Second, all of the objects related to state management have been removed, `StatefulMixin`, `WorkerFormData`, and `WorkerProperties`. These are no longer needed since the `App` class no longer needs to manage state.
+
+Since the `StatefulMixin` is no longer required, `App` can now descend directly from `WidgetBase`.
+
+{% instruction 'Remove `AppBase` and change `App` to descend from `WidgetBase`.' %}
+
+The new class declaration for `App` should be:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' line:8 %}
+
+Also, the property and methods within `App` that are setting and managing state can be removed.
+
+{% instruction 'Remove the following code from the `App` class.' %}
+
+{% include_codefile 'demo/initial/biz-e-corp/src/widgets/App.ts' lines:20-48 %}
+
+The final change to `App` is to update the `render` method to use the containers. Since the containers already know how to manage their state and respond to events, no properties need to be passed to them by the `App` widget.
+
+{% instruction 'Replace the `render` method with the following code.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:10-16 %}
+
+With this last change, the `App` class is now only nine lines of code. All of the state-management logic is still part of the application, but it has been factored out of the `App` class, which did not need it.
+
+Notice that the `WorkerForm` and `WorkerContainer` widgets were not changed at all. This is an important thing to keep in mind when designing widgets - a widget should never be concerned about where its properties come from, that is what containers are for. By keeping the containers and widgets separate, we have helped to ensure that they each have a narrowly defined set of responsibilities, making them easier to reuse, test, and maintain.
```

Comment:
"responsibilities, making them" -> "responsibilities, creating a cleaner separation of concerns, and making the widgets and containers"